lib/matplotlib/figure.py

```diff
@@ -731,6 +731,11 @@ def suptitle(self, t, *, x=.5, y=.98, **kwargs):
 
         >>> fig.suptitle('This is the figure title', fontsize=12)
         """
+        manual_position = ('x' in kwargs or 'y' in kwargs)
+
+        x = kwargs.pop('x', 0.5)
+        y = kwargs.pop('y', 0.98)
```

Comment:
Do these two lines really work? I may be naive, but neither 'x' nor 'y' belong to `**kwargs`, do they not?

Reply:
Blame @anntzer  :wink:   The new signature is `def suptitle(self, t, *, x=.5, y=.98, **kwargs):`  The little star in there means everything past the star is a kwarg; `x` and `y` are explicit kwargs...

Reply:
Oh, actually, I think  see what happened.  @anntzer put the new signature in, and I put the pop back in.  OTOH, now I'm confused if manual position ever comes back false now....

Reply:
Ok, neat ^^, but even if 'x' or 'y' are keyword-only arguments, do they belong to the special dictionary `**kwargs`?

Reply:
ping @anntzer is this OK, or is there a better way to check if `x` or `y` have been set?

Reply:
No, you have to use kwargs-popping in that case.

I'm a bit confused though (haven't followed the issue at all): if x and y are not set, are they always going to be 0.5 and 0.98, or is it that the constrained layout manager will move them around?  If the latter, then perhaps the default should just be x=y=None, as 0.5/0.98 don't really mean anything?  Although I guess they are still meaningful if the constrained layout manager is *not* in use...

Reply:
If x and y are not set, then they start life as 0.5 and 0.98.  If constrained layout is called, they will move.  If the user *does* set x or y then constrained_layout ignites the suptitle